low-level/framework/utils/ipmi_client.py

```diff
@@ -161,39 +187,68 @@ def get_fru_list_by_type(self, fru_list, sensor_id_map):
                 for fru in fru_detail}
         return sensor_id_map
 
-    def _run_command(self, command, out_file=subprocess.PIPE):
-        """executes commands"""
-        process = subprocess.Popen(command, shell=True, stdout=out_file, stderr=subprocess.PIPE)
-        result = process.communicate()
-        return result, process.returncode
-
-    def _run_ipmitool_subcommand(self, subcommand, grep_args=None, out_file=subprocess.PIPE):
-        """executes ipmitool sub-commands, and optionally greps the output"""
-
-        ipmi_tool = self.IPMITOOL
-
-        # A dummy file path check to select ipmi simulator if
-        # simulator is required, otherwise default ipmitool.
-        if os.path.exists("/tmp/activate_ipmisimtool"):
-            res, retcode = self._run_command(command=f"{self.IPMISIMTOOL} sel info")
-            if retcode == 0:
-                ipmi_tool = self.IPMISIMTOOL
-                logger.info("IPMI simulator is activated")
-
-        command = ipmi_tool + subcommand
-        if grep_args is not None:
-            command += " | grep "
-            if isinstance(grep_args, list):
-                grep_args_str = ""
-                for arg in grep_args:
-                    grep_args_str = "'{}' ".format(arg)
-                command += grep_args_str
-            else:
-                command += "'{}'".format(grep_args)
-        res, retcode = self._run_command(command, out_file)
+    def _run_ipmitool_subcommand(self, subcommand, grep_args=None):
```

Comment:
Please write a unittest for this function.

Reply:
already written.